auraed/src/runtime/cell_service/cells/cgroups/cpu_cpus.rs

```diff
@@ -1,29 +1,34 @@
-use fancy_regex::Regex;
+//use fancy_regex::Regex;
 
 use std::{
-    fmt::{Display, Formatter},
+    fmt::{Debug, Display, Formatter},
     ops::Deref,
 };
 
+use iter_tools::Itertools;
 use validation::{ValidatedField, ValidationError};
 
 #[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
-pub struct CpuCpus(String);
+pub struct CpuCpus(Vec<i32>);
 
 impl CpuCpus {
     #[cfg(test)]
-    pub fn new(cpu_cpus: String) -> Self {
+    pub fn new(cpu_cpus: Vec<i32>) -> Self {
         Self(cpu_cpus)
     }
 
-    pub fn into_inner(self) -> String {
+    pub fn into_inner(self) -> Vec<i32> {
         self.0
     }
+
+    pub fn as_string(&self) -> String {
```

Comment:
This should be implemented using the Display trait which gives a `to_string`: https://doc.rust-lang.org/std/string/trait.ToString.html

Reply:
i deliberately didn't because this is not for display; it's a conversion from one representation to another.  if it's idiomatic to use Display for this then i can, but i didn't think it was the right fit.

Reply:
I would probably name it something else then. Maybe `as_controller_format`? Or maybe it should be `to_controller_format` ([guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv))?

Reply:
Alternatively you could use a type alias, e.g. CpusStr, and implement the `From` trait so the conversion can be done with `.into()`.